client/src/game/reducers/cardDeductionReducer.ts

```diff
@@ -0,0 +1,343 @@
+          // empathy.
+          possibleCardsForEmpathy = possibleCardsForEmpathy.filter(
+            (possibility) =>
+              possibilityValid(possibility, possibilities, 0, cardCountMap),
+          );
+        }
+      }
+    } else if (playerIndex === card.location) {
+      possibleCardsForEmpathy = possibleCardsForEmpathy.filter((possibility) =>
+        possibilityValid(possibility, possibilities, 0, cardCountMap),
+      );
+    }
+    deck[card.order] = {
+      ...card,
+      possibleCards,
+      possibleCardsForEmpathy,
+    };
+  });
+}
+
+function getCardsToCalculate(
+  playerIndex: number,
+  ourPlayerIndex: number,
+  hands: ReadonlyArray<readonly number[]>,
+  deck: CardState[],
+  oldDeck: readonly CardState[],
+): CardState[] {
+  const cardsToCalculate: CardState[] = [];
+
+  hands.forEach((hand) => {
+    hand.forEach((order) => {
+      const card = deck[order];
+      if (
+        shouldCalculateCard(card, playerIndex, ourPlayerIndex, deck, oldDeck)
+      ) {
+        cardsToCalculate.push(card);
+      }
+    });
+  });
+  return cardsToCalculate;
+}
+
+function shouldCalculateCard(
+  card: CardState,
+  playerIndex: number,
+  ourPlayerIndex: number,
+  deck: CardState[],
+  oldDeck: readonly CardState[],
+) {
+  const cardPossibilitiesForPlayer = getCardPossibilitiesForPlayer(
+    card,
+    playerIndex,
+    ourPlayerIndex,
+  );
+
+  if (
+    card.revealedToPlayer[playerIndex] ||
+    cardPossibilitiesForPlayer.length === 1
+  ) {
+    // The player already knows what this card is.
+    return false;
+  }
+
+  const oldCard = oldDeck[card.order];
+
+  if (typeof oldCard === "undefined" || oldCard.location === "deck") {
+    // this is a newly drawn card and hasn't had any calculations yet.
+    return true;
+  }
+
+  // If the possibilities on the unknown cards don't change, then the result of our calculation
+  // won't change.  We only need to recalculate the card if the input (possibilities) changed.
+  return unknownCardPossibilitiesDifferent(
+    card.order,
+    deck,
+    oldDeck,
+    playerIndex,
+    ourPlayerIndex,
+  );
+}
+
+function getCardPossibilitiesForPlayer(
+  card: CardState,
+  playerIndex: number,
+  ourPlayerIndex: number,
+): ReadonlyArray<readonly [number, number]> {
+  if (card.location === playerIndex) {
+    // If this card is in the players hand, then use our best (empathy) guess.
+    return card.possibleCardsForEmpathy;
+  }
+  if (playerIndex === ourPlayerIndex || card.revealedToPlayer[playerIndex]) {
+    // This is revealed to the player or we are the requested player => just use our best knowledge.
+    return card.possibleCards;
+  }
+  // This is an unrevealed card not in the players hand but not revealed to them.
+  // That can happen with something like a detrimental character (such as 'Slow-Witted')
+  // or throw it in the hole.  We can't use our best (empathy) guess, because it might be in our own
```

Comment:
1) its throw it in a hole
2) capitalize it appropriately
3) one space between periods

Reply:
Done. I'm apparently behind the times with the "one space between periods" thing. Hopefully that habit won't be too tough to break.

Reply:
https://www.instructionalsolutions.com/blog/one-space-vs-two-after-period#:~:text=Nearly%20all%20style%20guides%20agree%20that%20one%20space%20is%20correct.&text=The%20Publication%20Manual%20of%20the,space%20in%20its%202019%20update.